src/vdaf/hits.rs

```diff
@@ -437,92 +437,87 @@ impl<I: Idpf<2, 2>> Aggregator for Hits<I> {
         })
     }
 
+    fn prepare_preprocess<M: IntoIterator<Item = Self::PrepareMessage>>(
+        &self,
+        inputs: M,
+    ) -> Result<Self::PrepareMessage, VdafError> {
+        let mut output: Option<Vec<I::Field>> = None;
+        let mut count = 0;
+        for data_share in inputs.into_iter() {
+            count += 1;
+            if let Some(ref mut data) = output {
+                if data_share.0.len() != data.len() {
+                    return Err(VdafError::Uncategorized(format!(
+                        "unexpected message length: got {}; want {}",
+                        data_share.0.len(),
+                        data.len(),
+                    )));
+                }
+
+                for (x, y) in data.iter_mut().zip(data_share.0.iter()) {
+                    *x += *y;
+                }
+            } else {
+                output = Some(data_share.0);
+            }
+        }
+
+        if count != 2 {
+            return Err(VdafError::Uncategorized(format!(
+                "unexpected message count: got {}; want 2",
+                count,
+            )));
+        }
+
+        Ok(HitsPrepareMessage(output.unwrap()))
+    }
+
     // TODO Fix this clippy warning instead of bypassing it.
     #[allow(clippy::type_complexity)]
-    fn prepare_step<M: IntoIterator<Item = HitsPrepareMessage<I::Field>>>(
+    fn prepare_step(
         &self,
         mut state: HitsPrepareStep<I::Field>,
-        inputs: M,
+        input: Option<HitsPrepareMessage<I::Field>>,
```

Comment:
I think we should be consistent about using either `HitsPrepareMessage` or `Self::PrepareMessage`. I lean toward `HitsPrepareMessage`.

Reply:
Done